src/m365/outlook/commands/mail/mail-searchfolder-add.ts

```diff
@@ -0,0 +1,92 @@
+import { MailSearchFolder } from '@microsoft/microsoft-graph-types';
+import { Logger } from '../../../../cli/Logger.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { entraUser } from '../../../../utils/entraUser.js';
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+import { z } from 'zod';
+import { globalOptionsZod } from '../../../../Command.js';
+import { zod } from '../../../../utils/zod.js';
+import { validation } from '../../../../utils/validation.js';
+
+const options = globalOptionsZod
+  .extend({
+    userId: zod.alias('i', z.string()
+      .refine(userId => validation.isValidGuid(userId), userId => ({
+        message: `'${userId}' is not a valid GUID.`
+      })).optional()),
+    userName: zod.alias('n', z.string()
+      .refine(userName => validation.isValidUserPrincipalName(userName), userName => ({
+        message: `'${userName}' is not a valid UPN.`
+      })).optional()),
+    folderName: z.string(),
+    messageFilter: z.string(),
+    sourceFoldersIds: z.string().transform((value) => value.split(',')).pipe(z.string().array()),
+    includeNestedFolders: z.boolean().optional()
+  })
+  .strict();
+
+declare type Options = z.infer<typeof options>;
+
+interface CommandArgs {
+  options: Options;
+}
+
+class OutlookMailSearchFolderAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MAIL_SEARCHFOLDER_ADD;
+  }
+
+  public get description(): string {
+    return `Creates a new mail search folder in the user's mailbox`;
+  }
+
+  public get schema(): z.ZodTypeAny | undefined {
+    return options;
+  }
+
+  public getRefinedSchema(schema: typeof options): z.ZodEffects<any> | undefined {
+    return schema
+      .refine(options => !options.userId !== !options.userName, {
+        message: 'Specify either userId or userName, but not both'
+      });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      let userId = args.options.userId;
+
+      if (args.options.userName) {
+        userId = await entraUser.getUserIdByUpn(args.options.userName);
+      }
+
+      if (args.options.verbose) {
+        await logger.logToStderr(`Creating a mail search folder in the mailbox of the user ${userId}...`);
+      }
```

Comment:
When doing this call, we can specify either the id or upn for the user, so this call to retrieve the ID is not needed.
```suggestion
      if (args.options.verbose) {
        await logger.logToStderr(`Creating a mail search folder in the mailbox of the user ${args.options.userId || args.options.userName}...`);
      }
```